### ISSUE TITLE
API suggestion for distinguishing between the binding of a property's get or set

### ISSUE THREAD

**Issue by ltcmelo** [dotnet/roslyn#26867]
It seems that currently there's no *direct* way, in the case of C# property accessors, to distinguish whether an identifier binds to the property's underlying `get` or `set` method symbol. Given that [such differentiation isn't straightforward](https://github.com/dotnet/csharplang/issues/1526), it might be worth to have an specific API for this task.

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani as well.   This would likely make sense in IOperation as well.

**Comment by mavasani** [comment]
> This would likely make sense in IOperation as well.

Yes, that is tracked by https://github.com/dotnet/roslyn/issues/25057.

**Comment by weltkante** [comment]
I'm surprised that there is still no API for this, I'm trying to implement accessor resolution for an analyzer (so the analyzer can check attributes on the getter/setter depending how its used) but it turns out to be really complex when you want exact results and not just some heuristic. In particular since Roslyn doesn't ever report the correct IPropertySymbol against which I'm binding (see #36403)

When adding an API please consider that there's also the third case of binding against the backing field, when writing to a readonly property in a constructor. When adding an API this case should be detectable from the Analyzer without performing additional complex checks.

**Comment by mavasani** [comment]
@weltkante Meanwhile, you can consider cloning http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Extensions/OperationExtensions.cs,23 for your analyzer package

**Comment by weltkante** [comment]
@mavasani sorry took a while until I could get around to look at the linked code

While the linked code does help somewhat it doesn't solve the actual problem, in particular binding against the backing field vs. the setter is proving tricky to resolve manually. For now I won't bother with it in my analyzer, but an actual API to figure out property symbol binding would be really nice.
